File: pinyougou-manager-web/src/main/webapp/js/controller/goodsController.js
 //控制层 
app.controller('goodsController' ,function($scope,$controller,goodsService,itemCatService){	
	
	$controller('baseController',{$scope:$scope});//继承
	
    //读取列表数据绑定到表单中  
	$scope.findAll=function(){
		goodsService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	}    
	
	//分页
	$scope.findPage=function(page,rows){			
		goodsService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
	
	//查询实体 
	$scope.findOne=function(id){				
		goodsService.findOne(id).success(
			function(response){
				$scope.entity= response;					
			}
		);				
	}
	
	//保存 
	$scope.save=function(){				
		var serviceObject;//服务层对象  				
		if($scope.entity.id!=null){//如果有ID
			serviceObject=goodsService.update( $scope.entity ); //修改  
		}else{
			serviceObject=goodsService.add( $scope.entity  );//增加 
		}				
		serviceObject.success(
			function(response){
				if(response.success){
					//重新查询 
		        	$scope.reloadList();//重新加载
				}else{
					alert(response.message);
				}
			}		
		);				
	}
	
	 
	//批量删除 
	$scope.dele=function(){			
		//获取选中的复选框			
		goodsService.dele( $scope.selectIds ).success(
			function(response){
				if(response.success){
					$scope.reloadList();//刷新列表
					$scope.selectIds=[];
				}						
			}		
		);				
	}
	
	$scope.searchEntity={};//定义搜索对象 
	
	//搜索
	$scope.search=function(page,rows){			
		goodsService.search(page,rows,$scope.searchEntity).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
	
	
	///////////////////////////////////////////////////从shop-web工程中拷贝过来
	//定义商品列表页面显示商品审核状态的数组变量
	$scope.status = ['未审核','已审核','审核未通过','已关闭'];
	
	//同理,将所有的分类信息查询出来,以分类的id当作数组的下标,分类的名称当作对应下标的值
	$scope.itemCatList =  []; 
	$scope.findItemCatList = function(){
		itemCatService.findAll().success(function(response){
			//注意这里是查询tb_item_cat,查询所有,返回的数据格式为:[{id:1,name="手机数码",parentId=0}]
			for(var i=0;i<response.length;i++){
				$scope.itemCatList[response[i].id] = response[i].name;
			}
			
		});
	}
	
	
	//运营商管理后台需要对商品进行审核,审核通过或者驳回其实就是修改商品的状态
	//更新状态
	$scope.updateStatus=function(status){
		//$scope.selectIds是在baseController中定义好的专门保存复选框选中的值的,注意页面需要给复选框绑定在baseController中写好的事件updateSelection
		goodsService.updateStatus( $scope.selectIds ,status).success(
			function(response){
				if(response.success){
					$scope.reloadList();//刷新页面
					
					//清空复选框选中而保存的值
					$scope.selectIds=[];
				}else{
					alert(response.message);
				}				
			}
		);		
	}
    
});	

File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/GoodsController.java
package com.pinyougou.manager.controller;
import java.security.KeyStore.PrivateKeyEntry;
import java.util.Arrays;
import java.util.List;

import javax.annotation.Resource;
import javax.jms.Destination;
import javax.jms.JMSException;
import javax.jms.Message;
import javax.jms.Session;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jms.core.JmsTemplate;
import org.springframework.jms.core.MessageCreator;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.alibaba.dubbo.config.annotation.Reference;
import com.alibaba.fastjson.JSON;
import com.github.pagehelper.MSUtils;
//import com.pinyougou.page.service.ItemPageService;
import com.pinyougou.pojo.TbGoods;
import com.pinyougou.pojo.TbItem;
import com.pinyougou.pojogroup.Goods;
//import com.pinyougou.search.service.ItemSearchService;
import com.pinyougou.sellergoods.service.GoodsService;

import entity.PageResult;
import entity.Result;
/**
 * controller
 * @author Administrator
 *
 */
@RestController
@RequestMapping("/goods")
public class GoodsController {

	@Reference
	private GoodsService goodsService;
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findAll")
	public List<TbGoods> findAll(){			
		return goodsService.findAll();
	}
	
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult  findPage(int page,int rows){			
		return goodsService.findPage(page, rows);
	}
	
	/**
	 * 要注意:这里为什么将新增商品的方法注释掉：因为该工程是运营商管理后台,不涉及商品的添加，商品都是由商家添加的
	 */
	/*@RequestMapping("/add")
	public Result add(@RequestBody TbGoods goods){
		try {
			goodsService.add(goods);
			return new Result(true, "增加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "增加失败");
		}
	}*/
	
	/**
	 * 修改
	 * @param goods
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody Goods goods){
		try {
			goodsService.update(goods);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
	}	
	
	/**
	 * 获取实体
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public Goods findOne(Long id){
		return goodsService.findOne(id);		
	}
	
	
	@Autowired
	private Destination queueSolrDeleteDestination;
	
	@Autowired
	private Destination topicPageDeleteDestination;
	/**
	 * 批量删除
	 * @param ids
	 * @return
	 */
	@RequestMapping("/delete")
	public Result delete(final Long [] ids){
		try {
			goodsService.delete(ids);
			//从索引库中删除
			//itemSearchService.deleteByGoodsIds(Arrays.asList(ids));
			
			//异步调用ActiveMQ 删除索引库中的数据
			JmsTemplate.send(queueSolrDeleteDestination, new MessageCreator() {
				
				@Override
				public Message createMessage(Session session) throws JMSException {
					return session.createObjectMessage(ids);//Long是实现了可序列化接口的
				}
			});
			//删除每个服务器上的商品详细页
			JmsTemplate.send(topicPageDeleteDestination, new MessageCreator() {
				
				@Override
				public Message createMessage(Session session) throws JMSException {
					return session.createObjectMessage(ids);
				}
			});
			
			return new Result(true, "删除成功"); 
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "删除失败");
		}
	}
	
		/**
	 * 查询+分页
	 * @param brand
	 * @param page
	 * @param rows
	 * @return
	 */
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbGoods goods, int page, int rows  ){
		return goodsService.findPage(goods, page, rows);		
	}
	/**
	 * 修改商品状态
	 * @param ids
	 * @param status
	 * @return
	 */
	@RequestMapping("/updateStatus1")
	public Result updateStatus1(Long[] ids, String status){
		try {
			goodsService.updateStatus(ids, status);
			return new Result(true, "成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "失败");
		}		
	}
	
	//@Reference(timeout=100000)
	//private ItemSearchService itemSearchService;
	
	@Autowired
	private JmsTemplate JmsTemplate;
	
	/**
	 * 注意这里名字和配置文件中的保持一致,@Autowired按类型注入,容器中到时候Destination类型的不止一个,
	 * 但是如果这里的变量名和容器中bean的名称保持一致，则autowired可以在按照类型注入的基础之上按照名称注入queueSolrDestination
	 * 
	 * 当然:我们可以直接使用@Resource注解按照名称注入
	 */
	@Autowired 
	private Destination queueSolrDestination;
	
	@Resource(name="topicPageDestination")
	private Destination topicPageDestination;
	/**
	 * 修改商品状态
	 * 然后如果商品状态为：审核通过,则需要将该商品的数据导入到索引库
	 * @param ids
	 * @param status
	 * @return
	 */
	@RequestMapping("/updateStatus")
	public Result updateStatus(Long[] ids, String status){
		try {
			goodsService.updateStatus(ids, status);
			if("1".equals(status)){//如果是审核通过 
				//得到需要导入的SKU列表
				List<TbItem> itemList = goodsService.findItemListByGoodsIdListAndStatus(ids, status);
				//导入到solr
				//itemSearchService.importList(itemList);	
				
				//利用ActiveMQQueue点对点将商品信息导入索引库
				System.out.println("导入到索引库之前查询到的数据:"+itemList);
				final String jsonString = JSON.toJSONString(itemList);//转换为json传输
				System.out.println("导入到索引库之前转换为json:"+jsonString);
				JmsTemplate.send(queueSolrDestination, new MessageCreator() {
					
					@Override
					public Message createMessage(Session session) throws JMSException {
						return session.createTextMessage(jsonString);
					}
				});
				
				//****生成商品详细页
				/*for(Long goodsId:ids){
					itemPageService.genItemHtml(goodsId);
				}*/
				
				//利用ActiveMQTopic发布订阅的方式将商品id传输,然后生成商品详情页
				System.out.println("-----审核时生成商品详情页:得到的topic"+topicPageDestination);
				for(final Long goodsId:ids){
					JmsTemplate.send(topicPageDestination, new MessageCreator() {
						
						@Override
						public Message createMessage(Session session) throws JMSException {
							return session.createTextMessage(goodsId+"");
						}
					});
				}
				
			}	
			return new Result(true, "成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "失败");
		}		
	}
	
	//@Reference(timeout=40000)
	//private ItemPageService itemPageService;
	
	/**
	 * 调用itemPageService生成静态html
	 * @param goodsId
	 */
	/*@RequestMapping("/genHtml")
	public void genHtml(Long goodsId){
		itemPageService.genItemHtml(goodsId);
	}*/
}
